lib/Service/FilesService.php

```diff
@@ -493,14 +493,18 @@ private function generateFilesDocumentFromFile(string $viewerId, Node $file): Fi
 		}
 
 		$document->setModifiedTime($file->getMTime());
-
 		$stat = $file->stat();
-		$document->setMore(
-			[
-				'creationTime' => $this->getInt('ctime', $stat),
-				'accessedTime' => $this->getInt('atime', $stat)
-			]
-		);
+
+		if (is_array($stat)) {
+			$document->setMore(
+				[
+					'creationTime' => $this->getInt('ctime', $stat),
+					'accessedTime' => $this->getInt('atime', $stat)
+				]
+			);
+		} else {
+			$this->log(2, 'stat() on File #' . $file->getId() . ' is not an array: ' . json_encode($stat));
```

Comment:
Where is this log function defined?

Reply:
Yeah, I have the same issue and question. Where I can define this? I have these libs under /var/www/nextcloud/apps/fulltextsearch_elasticsearch/lib/service: ConfigService.php         MiscService.php           SettingsService.php
IndexMappingService.php   SearchMappingService.php
IndexService.php          SearchService.php

Reply:
defined in a Trait: https://github.com/nextcloud/files_fulltextsearch/blob/21268d75c6bb74c3649153582d70a2ccadc0567e/lib/Service/FilesService.php#L83

Reply:
And what is the '2' magic value? There is no constant for this?
Why are you not using the logger through DI as other applications are doing?
Not a blocker for this PR of course, something to look at later.